servant-server/src/Servant/Server/Internal/ErrorFormatter.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE DataKinds #-}
+
+-- | TODO: more documentation for this module
+module Servant.Server.Internal.ErrorFormatter
+  where
+
+import           Data.String.Conversions
+                 (cs)
+import           Data.Typeable
+
+import           Servant.API.ContentTypes
+import           Servant.Server.Internal.Context
+import           Servant.Server.Internal.ServerError
+
+-- | 'Context' that contains default formatters for all error types.
+--
+-- Default formatters will just return HTTP 400 status code with error
+-- message as response body.
+type DefaultErrorFormatters = '[BodyParseErrorFormatter, URLParseErrorFormatter, HeaderParseErrorFormatter]
```

Comment:
Would it mabe not make sense to have:


```
data ErrorConfig = ErrorConfig 
   { bodyParseErrorFormatter :: ErrorFormatter
   , ...
  }
```
And then have that be the only thing in the context? (As I mentioned in another context in this PR, the constructor would not be exposed except in an `Internal` module, and only the record fields would. In this case additionally a `defaultErrorConfig :: ErrorConfig` would also be exposed.) It seems like that would help us with backwards-compatibility in the future.

Reply:
I wanted to make different error types independent, following the second option from your initial proposal. I do not have a strong preference though.

Reply:
Would you be okay implementing the single-datatype approach then? I think the difference in how many breaking changes we'll have with one vs the other is pretty significant. (Sorry for the extra work!)